src/integrationTest/java/com/mongodb/hibernate/query/select/SimpleSelectQueryIntegrationTests.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.query.select;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.mongodb.hibernate.junit.MongoExtension;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.util.List;
+import java.util.function.Consumer;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.query.SelectionQuery;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.jspecify.annotations.Nullable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@SessionFactory(exportSchema = false)
+@DomainModel(annotatedClasses = SimpleSelectQueryIntegrationTests.Contact.class)
+@ExtendWith(MongoExtension.class)
+class SimpleSelectQueryIntegrationTests {
+
+    @BeforeEach
+    void beforeEach(SessionFactoryScope scope) {
+        scope.inTransaction(session -> List.of(
+                        new Contact(1, "Bob", 18, Country.USA),
+                        new Contact(2, "Mary", 35, Country.CANADA),
+                        new Contact(3, "Dylan", 7, Country.CANADA),
+                        new Contact(4, "Lucy", 78, Country.CANADA),
+                        new Contact(5, "John", 25, Country.USA))
+                .forEach(session::persist));
+    }
+
+    @Test
+    void testComparisonByEQ(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country = :country",
```

Comment:
Does HQL allow the parameter and column name to be reversed, e.g.

```
from Contact where :country = country
```

If so, let's add a test for it to ensure that it's also supported even though it's probably much rarer.

Reply:
HQL seems to support this. Good catch! I need to modify my PR to accommodate it. Currently its logic assumes the left hand side expression is the field path, which is incorrect.

Reply:
I refactored the code to support this, but not sure it is enough. Currently I assumed at least one of the left and right sides refer to a column (`Assignable` interface in Hibernate); and throw exception if not. Like `where 1 = 2`. If we need to support such ad-hoc relational predicate, further nontrivial code refactoring is required. How do you think? Could we assume at least one side refers to a field?

Reply:
Just found my refactoring is defective in that it doesn't revert operator for the following case:
```
from Contact where :age > age
```
My current logic will treat it as the same as:
```
from Contact where age > :age
```
which is totally wrong.
Need to work further.

Reply:
I solved the issue in latest commit by tapping into Hibernate's `ComparisonOperator`'s `invert` feature, so if field path shows up on the right side, we need to use the `inverted` operator instead.

Seems MongoDB's comparison operator only supports one field and one value (order-agnostic as above). So the following two kinds of HQL should end up with `FeatureNotSupportedException`:
```
where 1 = 2
```
```
where field1 = field2
```
I added the validation logic to ensure only field and value pair is supported.